enterprise/server/util/oci/oci_test.go

```diff
@@ -682,76 +683,78 @@ func TestResolve_WithCache(t *testing.T) {
 			},
 		},
 	} {
-		writeLayers := false
 		readLayers := false
 		for _, writeManifests := range []bool{false, true} {
 			for _, readManifests := range []bool{false, true} {
-				name := tc.name + fmt.Sprintf(
-					"/write_manifests_%t_read_manifests_%t_write_layers_%t_read_layers_%t",
-					writeManifests,
-					readManifests,
-					writeLayers,
-					readLayers,
-				)
-				t.Run(name, func(t *testing.T) {
-					te := setupTestEnvWithCache(t)
-					flags.Set(t, "executor.container_registry_allowed_private_ips", []string{"127.0.0.1/32"})
-					flags.Set(t, "executor.container_registry.use_cache_percent", 100)
-					flags.Set(t, "executor.container_registry.write_manifests_to_cache", writeManifests)
-					flags.Set(t, "executor.container_registry.read_manifests_from_cache", readManifests)
-					counter := atomic.Int32{}
-					registry := testregistry.Run(t, testregistry.Opts{
-						HttpInterceptor: func(w http.ResponseWriter, r *http.Request) bool {
-							if r.Method == http.MethodGet && r.URL.Path != "/v2/" {
-								counter.Add(1)
+				for _, writeLayers := range []bool{false, true} {
```

Comment:
this is getting pretty deeply nested; it might be clearer to create a `testCase` struct and build up a list of test cases using the triple-nested loop, then run the flat list of test cases in another flat loop.